### PR TITLE
[BACK-1737] Delete user profiles asynchronously

### PR DESCRIPTION
**Pull request by toddkazakov**

### CODE REVIEW

File: lib/routes/seagullApi.js [hunk 1]
```diff
@@ -302,10 +304,6 @@ module.exports = function (crudHandler, userApiClient, gatekeeperClient, metrics
               }
             });
             userProfiles = _.flatten(results);
-
-            if (userProfiles.length !== userIds.length) {
-              throw new Error('Received different number of results than we queried for');
-            }
           } catch (error) {
             res.send(500);
```
Comment (derrickburns) [inline lib/routes/seagullApi.js:308]: Is it possible to get here now?  If so, we should log an error message.  If not, we should delete this code path.

**Review by derrickburns — CHANGES_REQUESTED** [review]
Small change request.

File: lib/routes/seagullApi.js (hunk 1 above)
Comment (toddkazakov) [reply lib/routes/seagullApi.js:308]: It's possible if `const users = await getUsers(usersChunk);` throws an exception - I'll get that fixed.

Comment (toddkazakov) [reply lib/routes/seagullApi.js]: This was resolved after rebasing master

**Review by derrickburns — APPROVED** [review]
Lgtm

**Review by derrickburns — APPROVED** [review]
LGTM

**Pull request merged**